### ISSUE TITLE
Expo-font `useFonts` hook freezes Android when built through `eas build`

### ISSUE THREAD

**Issue by rodrigohgpontes** [expo/expo#17876]
### Summary

When I use the hook `useFonts` from the package "expo-font" to load custom fonts in my app and I use `eas build` to build for Android, the app freezes on physical devices on the splash screen. It has this issue when building both APK and AAB.

When running through Expo Go, it works fine. When using classic `expo build`, it works fine. When building for iOS, it works fine. If I remove the use of `useFonts` from the code below, it works fine.

The issue was also happening when I used the "@expo-google-fonts" packages instead of local font files.

**App.js**
```
import React from "react";
import { ActivityIndicator } from "react-native";
import IndexJS from "./index";
import { useFonts } from "expo-font";

export default function App() {
  const [fontsLoaded] = useFonts({
    "Delius_400Regular": require("./src/assets/fonts/Delius-Regular.ttf"),
    "GloriaHallelujah_400Regular": require("./src/assets/fonts/GloriaHallelujah-Regular.ttf"),
  });

  if (!fontsLoaded) {
    return <ActivityIndicator size="small" color="black" animating={fontsLoaded} />;
  }

  return <IndexJS />;
}
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

45

### Environment

expo-env-info 1.0.3 environment info:
    System:
      OS: Linux 5.17 Fedora Linux 36 (Workstation Edition)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 16.15.0 - /usr/bin/node
      Yarn: 1.22.18 - /usr/local/bin/yarn
      npm: 8.10.0 - /usr/local/bin/npm
    npmPackages:
      expo: ~45.0.0 => 45.0.4 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      eas-cli: 0.53.1
      expo-cli: 5.4.9
    Expo Workflow: managed

### Reproducible demo

**package.json**
```
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "eas:build:ios": "sudo eas build --profile development --platform ios"
  },
  "dependencies": {
    "@react-native-masked-view/masked-view": "0.2.6",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.6.2",
    "@react-navigation/stack": "^6.2.1",
    "eas-cli": "0.53.1",
    "expo": "~45.0.0",
    "expo-cli": "^5.4.9",
    "expo-contacts": "~10.2.0",
    "expo-device": "~4.2.0",
    "expo-firebase-recaptcha": "~2.2.0",
    "expo-font": "~10.1.0",
    "expo-image-picker": "~13.1.1",
    "expo-linear-gradient": "~11.3.0",
    "expo-notifications": "~0.15.2",
    "expo-status-bar": "~1.3.0",
    "expo-updates": "~0.13.2",
    "firebase": "8.2.3",
    "lottie-react-native": "5.0.1",
    "moment": "^2.29.3",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-native": "0.68.2",
    "react-native-animated-loader": "^0.0.9",
    "react-native-confirmation-code-field": "^7.3.0",
    "react-native-country-picker-modal": "^2.0.0",
    "react-native-gesture-handler": "~2.2.1",
    "react-native-reanimated": "~2.8.0",
    "react-native-responsive-screen": "^1.4.2",
    "react-native-safe-area-context": "4.2.4",
    "react-native-screens": "~3.11.1",
    "react-native-svg": "12.3.0",
    "react-native-timeago": "^0.5.0",
    "react-native-web": "0.17.7",
    "react-native-webview": "11.18.1"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9"
  },
  "private": true
}
```

**Logs when running `adb logcat *:S ReactNative:V ReactNativeJS:V`**

```
- waiting for device -
--------- beginning of crash
--------- beginning of system
--------- beginning of main
06-15 05:53:10.459   602   718 I ReactNativeJS: Running "main
06-15 05:53:10.631   602   718 E ReactNativeJS: Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
06-15 05:53:10.631   602   718 E ReactNativeJS: 
06-15 05:53:10.631   602   718 E ReactNativeJS: This error is located at:
06-15 05:53:10.631   602   718 E ReactNativeJS:     in Unknown
06-15 05:53:10.631   602   718 E ReactNativeJS:     in Unknown
06-15 05:53:10.631   602   718 E ReactNativeJS:     in RCTView
06-15 05:53:10.631   602   718 E ReactNativeJS:     in Unknown
06-15 05:53:10.631   602   718 E ReactNativeJS:     in RCTView
06-15 05:53:10.631   602   718 E ReactNativeJS:     in Unknown
06-15 05:53:10.631   602   718 E ReactNativeJS:     in b
```

I also logged the `error` from `const [fontsLoaded, error] = useFonts()`, but it just returns `null`.

**Comment by brentvatne** [comment]
the error message you are getting is telling you that somewhere in your app you are trying to render an object rather than a react component. i recommend following the above advice for generating a minimal reproducible example and you will likely find the mistake in your code

**Comment by rodrigohgpontes** [comment]
Thanks, @brentvatne.  I was able to fix the problem. You might want to look at it, even though you were correct that it's not an expo-font issue.

This issue is correctly closed, as it turns out that the `useFonts` was not the issue. Although removing it would fix it for some mysterious reason. The fix is mysterious to me also, btw.

As a reminder, the error I got was `Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.`

And I had this `App.js`:
```
import React from "react";
import { ActivityIndicator } from "react-native";
import Index from "./index";
import { useFonts } from "expo-font";

export default function App() {
  const [fontsLoaded] = useFonts({
    Delius_400Regular: require("./src/assets/fonts/Delius-Regular.ttf"),
    GloriaHallelujah_400Regular: require("./src/assets/fonts/GloriaHallelujah-Regular.ttf"),
  });

  if (!fontsLoaded) {
    return <ActivityIndicator size="small" color="black" animating={fontsLoaded} />;
  }

  console.log('>>> will load Index', typeof Index); // added for debugging
  return <Index />;
}
```

I added a bunch of `console.log` around the code and I found out that in the one above, in the APK build, the component `Index` was coming as an `object`  (instead of the `function` it is).

The export/import was correct. I checked. It is also clear that it's correct since the same code would work fine with Expo Go, the iOS build, and when using the classic `expo build`. In all those cases, the log above would show `Index` as a function. Only when using `eas build` to build Android, it would be an object.

I do not have the slightest idea of why that happened, but it was like this.

The fix was to stop using an `<Index>` component and move all its functionality (basically just adding the navigation container) to `App.js` itself. It worked fine after that only change.
